test/cql-pytest/test_describe.py

```diff
@@ -552,6 +552,52 @@ def test_whitespaces_in_table_options(cql, test_keyspace, scylla_only):
         desc = "\n".join([d.create_statement for d in cql.execute(f"DESC TABLE {tbl}")])
         assert re.search(regex, desc) == None
 
+# Randomly create many UDTs, with dependencies between them 
+# and validate if describe displays them in correct order.
+# UDTs should be sorted topologically, meaning if UDT `a`
+# is used to create UDT `b`, then `a` should be before `b`.
+def test_udt_sorting(scylla_only, cql, test_keyspace, random_seed):
+    repeat = 10 # test it randomized, so repeat it inside the test to increase coverage
+    max_selected_udts = 5 # max number of UDTs that can be used to create a new UDT
+
+    for _ in range(repeat):
+        created_udts = []
+        udts_cnt = 25
+        udt_backup = []
+
+        with ExitStack() as stack:
+            # Create `udts_cnt` many UDTs
+            while len(created_udts) < udts_cnt:
+                select_cnt = random.randint(0, min(max_selected_udts, len(created_udts))) # how many UDTs we want to use to create a new one
+                udt_body = "(a int)"
+                if select_cnt != 0:
+                    selected_udts = random.sample(created_udts, select_cnt) # randomly select `select_cnt` already created UDTs
+                    udt_body = create_udt_body(selected_udts)
+                
+                udt = stack.enter_context(new_type(cql, test_keyspace, udt_body))
+                created_udts.append(udt)
+            
+            # Describe all UDTs, iterate over all of them and verify the order is correct .
+            # (If UDT `u` is used create UDT `w`, `u` has to be described before `w`)
+            desc = cql.execute(f"DESC KEYSPACE {test_keyspace}").all()
+            visited_udts = set()
+            for row in desc[1:]: # skip keyspace entry
+                if row.type == "type":
+                    # Extract UDTs used in current UDT
+                    # Names of all UDTs begin with `unique_name_prefix`
+                    for it in re.finditer(f"<({unique_name_prefix}[^\s\>]+)>", row.create_statement):
+                        assert it.group(1) in visited_udts
+                    visited_udts.add(row.name)
+            
+            assert len(visited_udts) == len(created_udts)
+            udt_backup = desc
+
+        # Recreate all UDTs from the description in a new keyspace.
+        with new_test_keyspace(cql, "WITH REPLICATION = {'class': 'NetworkTopologyStrategy', 'replication_factor': 1}") as keyspace:
```

Comment:
I think it would make sense to create the keyspace at the top level and nest the `for _ in range(repeat):` loop inside it. We will only have to create the keyspace once and not 10 times which should speed up the test.